File: youpinhui-sellergoods-interface/src/main/java/com/youpinhui/sellergoods/service/ItemCatService.java
package com.youpinhui.sellergoods.service;
import java.util.List;
import com.youpinhui.pojo.TbItemCat;

import  com.youpinhui.entity.PageResult;
/**
 * provider interface
 * @author Administrator
 *
 */
public interface ItemCatService {

	/**
	 * find all
	 * @return
	 */
	public List<TbItemCat> findAll();
	
	
	/**
	 * query according pagination 
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * Add
	*/
	public void add(TbItemCat itemCat);
	
	
	/**
	 * update
	 */
	public void update(TbItemCat itemCat);
	

	/**
	 * get one by id
	 * @param id
	 * @return
	 */
	public TbItemCat findOne(Long id);
	
	
	/**
	 * batch delete
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * fuzzy search 
	 * if the object  is not null , get the parameter from it as the search condition.
	 * @param pageNum current page number
	 * @param pageSize size of the page to display
	 * @return
	 */
	public PageResult findPage(TbItemCat itemCat, int pageNum,int pageSize);


	public List<TbItemCat> findByParentId(Long parentId);
	
}

File: youpinhui-manager-web/src/main/webapp/js/service/itemCatService.js
//Sevice
app.service('itemCatService',function($http){
	    	
	 //test case , get all the data return a Json
	this.findAll=function(){
		return $http.get('../itemCat/findAll.do');		
	}
	//pagination
	this.findPage=function(page,rows){
		return $http.get('../itemCat/findPage.do?page='+page+'&rows='+rows);
	}
	//find one object
	this.findOne=function(id){
		return $http.get('../itemCat/findOne.do?id='+id);
	}
	//add 
	this.add=function(entity){
		return  $http.post('../itemCat/add.do',entity );
	}
	//update 
	this.update=function(entity){
		return  $http.post('../itemCat/update.do',entity );
	}
	//batch delete
	this.dele=function(ids){
		return $http.get('../itemCat/delete.do?ids='+ids);
	}
	//search
	this.search=function(page,rows,searchEntity){
		return $http.post('../itemCat/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
	
	//findByParentId
	this.findByParentId=function(parentId){
		return $http.post('../itemCat/findByParentId.do?parentId='+parentId);
	}    	
	
	
	
	
});
